packages/block-editor/src/components/rich-text/use-input-rules.js

```diff
@@ -108,10 +108,12 @@ export function useInputRules( props ) {
 
 			const transformed = formatTypes.reduce(
 				( accumlator, { __unstableInputRule } ) => {
-					if ( __unstableInputRule ) {
+					if (
+						__unstableInputRule &&
+						event.target.tagName !== 'CODE'
+					) {
 						accumlator = __unstableInputRule( accumlator );
```

Comment:
Why are you disabling _all_ input rules when the the rich text wrapper is `<code>`? Maybe check `formatType.tagName` instead of hardcoding it?

Reply:
Also, I think technically code tags can be nested (valid HTML). I guess you're disabling all input rules because you generally don't want any input rules to be run in the code block. In that case, please add a prop to RichText similar to `__unstableAllowPrefixTransformations` and add it to the code block instead of hardcoding it here.

Reply:
Maybe `disableInputRules`.